src/codegen/ast.lisp

```diff
@@ -207,11 +210,13 @@ coalton symbols (`parser:identifier`)"
 
 (defstruct (node-application (:include node))
   "Function application (f x)"
-  (rator (util:required 'rator) :type node      :read-only t)
-  (rands (util:required 'rands) :type node-list :read-only t))
+  (properties (util:required 'properties) :type util:symbol-list :read-only t)
```

Comment:
This should have a comment with documentation above it. What are some things we currently expect may be in `properties`? What is it supposed to do?

In addition, the type would just be `list`, or you could define a new type `plist`. The type `symbol-list` only incidentally works because `t` and `nil` are symbols.